trace/trace.go

```diff
@@ -238,16 +246,20 @@ func (s *Span) Send() {
 	s.childrenLock.Lock()
 	for _, child := range s.children {
 		if !child.IsAsync() {
-			if !child.isSent {
+			child.sendLock.RLock()
+			isChildSent := child.isSent
+			child.sendLock.RUnlock()
+			if !isChildSent {
 				child.AddField("meta.sent_by_parent", true)
```

Comment:
it seems like it could still be possible for this field to be set on a child which was not sent by a parent? I'm not sure how important that is if it possible though 🤔

Reply:
Yeah this is not perfect, but holding the lock here will deadlock the `Send` method. I think it might require a new method that sets the field while under the send lock.